.doks/doks-cluster/cluster.tf

```diff
@@ -5,43 +5,23 @@ resource "digitalocean_kubernetes_cluster" "codacy_k8s" {
 
   node_pool {
     name       = "codacy-doks-pool-${terraform.workspace}"
-    size       = var.node_type
-    node_count = var.num_nodes
+    size       = "s-2vcpu-2gb"
+    auto_scale = true
+    node_count = 1
```

Comment:
The digital ocean kubernetes cluster terraform resource isn't great. It kind of enforces you to always have a default node_pool defined inside the cluster. The problem is that once that node pool is created, any change to it will cause terraform to want to destroy the cluster and provision a new cluster, instead of just a new node pool. 
Others have found the same problem [here](https://github.com/terraform-providers/terraform-provider-digitalocean/issues/350) and [here](https://github.com/terraform-providers/terraform-provider-digitalocean/issues/303).
The solution was to get the node pool size down to zero, then manually delete it through digital ocean's UI. This enforces the default node pool never to exist. Then we created another extra node pool (see below).